구글 회원 가입 및 로그인 테스트 코드 추가 및 리팩토링

inshining opened the pull request:
TODO: 참고해서 코멘트 부탁해요~

`Commits`에 들어가면 커밋마다 추가 설명을 달아 두었으니 함께 참고하세요~

inshining committed:
➕[ADD] : encrypt test

inshining committed:
➕[ADD] : TokenProviderTest

inshining committed:
➕[ADD] : 구글 회원가입시 이미 존재하는 이메일인지 확인

inshining committed:
♻️[REFACTOR] : userServiceTest의 중복된 변수 정리

inshining committed:
♻️[REFACTOR] : 로그인 유저 객체 생성을 User 모델로 이관
기존에 유저 객체 생성을 Service 레이어에서 진행되었다. 서비스 레이어에서 비지니스 로직을 다뤄야 한다는 생각에 유저 객체 생성은 유저 모델 클래스로 옮겼다.

inshining committed:
♻️[REFACTOR] : 토큰 만드는 메소드 추출
이메일 로그인, 구글 로그인, 리프레시 토큰 3가지 메소드에서 모두 동일한 코드가 반복되어서 코드 중복성을 줄이기 위해서 토큰 만드는 메소드 추출

inshining committed:
➕[ADD] : 구글 로그인 테스트 코드 추가

inshining commented on src/main/java/com/ddoddii/resume/service/UserService.java:
```diff
@@ -125,15 +115,25 @@ public UserAuthResponseDTO emailLogin(UserEmailLoginRequestDTO userLoginRequestD
+                userId(saveUser.getId())
+                .name(saveUser.getName())
+                .email(saveUser.getEmail())
+                // TODO: Enum 값이 잘못 설정되어 있음.
```
여기 확인 좀

ddoddii replied:
수정했습니다 !

inshining commented on src/main/java/com/ddoddii/resume/service/UserService.java:
```diff
@@ -125,15 +115,25 @@ public UserAuthResponseDTO emailLogin(UserEmailLoginRequestDTO userLoginRequestD
+                .email(saveUser.getEmail())
+                // TODO: Enum 값이 잘못 설정되어 있음.
+                // LoginType.GOOGLE 로 설정해야 함.
                 .loginType(LoginType.EMAIL)
```
```java
.loginType(user.getLoginType())
```
으로 수정할 수 없을까요? 
User 모델에 속성값이 있는데. 여기서 하드코딩하는 이유가 있나요??

ddoddii replied:
맞네용 수정했습니다

inshining commented on src/main/java/com/ddoddii/resume/service/UserService.java:
```diff
@@ -156,19 +156,14 @@ public UserAuthResponseDTO googleLogin(UserGoogleLoginRequestDTO userGoogleLogin
             UserDTO newUser = googleSignUp(userGoogleLoginRequestDTO);
+
+            // TODO: 무엇을 검증하기 위한 코드인지 설명이 필요.
+            // googleSignUp 메소드이 성공적으로 동작했다면 디비에 유저 정보가 저장되는 것이 보장되어 있음.
```
유저 객체가 필요하기 때문에 findById를 호출하는거죠?

ddoddii replied:
넹 더 좋은 방법이 있다면....알려주세요.......

inshining replied:
당장은 이렇게 유지하시죠...

inshining committed:
➕[ADD] : refresh token 토큰 생성 테스트코드

inshining committed:
😺[CHORE] : loginType을 유저에서 가져오기

inshining committed:
➕[ADD] : UserGoogleLoginRequestDTO에 Valid 추가

inshining committed:
➕[ADD] : Valid 추가

inshining committed:
⛑[FIX] :  RuntimeException -> NotExistIdException으로 변경
GlobalExceptionHandler에 Runtimexception 없어서 이 에러 발생시 5xx 에러 남. 현재 임의로  `NotExistIdException`으로 변경해둠.

inshining committed:
➕[ADD] : controller 테스트 작성

inshining committed:
➕[ADD] : testcontainer mysql 설정

inshining committed:
➕[ADD] : user 통합 테스트 추가

inshining committed:
Merge branch 'develop' into test/user

inshining commented:
# 주요 변경 사항 
### loginType 리턴할 때, user 객체에서 받아옴. (기존에 하드코딩으로 되어 있었음) [코드](https://github.com/ddoddii/interview-plus-spring-server/pull/6/commits/1557b95a1b8065330da5fb38aa73e1423a662e5f)

### Vaild 보강 : controller에서 @Valid 추가, @NotBlank, Email 추가 

### 예외처리 변경: [참고](https://github.com/ddoddii/interview-plus-spring-server/pull/6/commits/f7157b35dcde06e93712e5b1f829b8fcaa42dcc9)

### 컨트롤러 테스트 추가 

### 테스트 컨테이너 추가 
통합 테스트를 구축하기 위해서 데이터베이스 필요. 
현재 데이터베이스는 테스트용으로 적절하지 않음. (원치 않는 데이터 삭제 가능성, 멱등성 필요) 
h2와 같은 메모리 디비도 적절하지 않았음. 실제로 사용하는 디비는 mysql이기 때문에 디비 타입이 달라서 생기는 문제 발생 소지있음.

-> [테스트컨테이너](https://testcontainers.com/) 사용
테스트컨테이너는 디비 도커 이미지 사용해서 테스트용으로 디비를 연결해주는 라이브러리임 

### 유저 통합 테스트 완료

inshining commented:
코드보고 Approve 하면 제가 merge 할게요~

inshining committed:
Merge branch 'develop' into test/user
